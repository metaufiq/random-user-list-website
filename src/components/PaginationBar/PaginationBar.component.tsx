import { Props } from "./PaginationBar.component.types";

const _renderActiveButton = (index: number) => (
  <li key={index}>
    <a 
      href="#" 
      aria-current="page" 
      className="py-2 px-3 text-blue-600 bg-blue-50 border border-gray-300 hover:bg-blue-100 hover:text-blue-700 dark:border-gray-700 dark:bg-gray-700 dark:text-white"
      >
        {index}
    </a>
  </li>
)

const _renderInActiveButton = (index: number) => (
  <li key={index}>
    <a 
      href="#" 
      className={`py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white`}
      >
        {index}
    </a>
  </li>
)

const _renderPreviousButton = (props: Props) => {
  return (
    <li>
    <a href="#" className="py-2 px-3 ml-0 leading-tight text-gray-500 bg-white rounded-l-lg border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white">Previous</a>
    </li>
  )
}

const _renderNextButton = (props: Props) => (
  <li>
    <a href="#" className="py-2 px-3 leading-tight text-gray-500 bg-white rounded-r-lg border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white">Next</a>
  </li>
)

const PaginationBar= (props: Props) => {
  const pageIndexes = new Array(props.totalPages).fill(0);
  
  return (
    <nav aria-label="Page navigation example">
      <ul className="inline-flex -space-x-px">
        {props.currentIndex != 0 && _renderPreviousButton(props)}
        {
          pageIndexes.map((data,index)=>index === props.currentIndex - 1 ? _renderActiveButton(index+1) : _renderInActiveButton(index+1))
        }
        {props.currentIndex !== props.totalPages-1 && _renderNextButton(props)}
      </ul>
    </nav>
  )
}

export default PaginationBar;