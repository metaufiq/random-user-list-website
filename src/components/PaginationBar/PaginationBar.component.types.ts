export type Props = {
  currentIndex: number,
  totalPages: number,
  onChange: (index:number)=> VoidFunction
}