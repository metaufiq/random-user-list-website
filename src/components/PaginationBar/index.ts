import PaginationBar from "./PaginationBar.component";

export default PaginationBar;