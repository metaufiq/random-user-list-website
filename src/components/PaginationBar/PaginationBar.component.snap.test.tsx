import { assertSnapshots } from "../../../test";
import PaginationBar from ".";
import { Props } from "./PaginationBar.component.types";

describe('PaginationBarSnapshot', () => {
  const mockProps:Props = {
    currentIndex: 1,
    onChange: (index)=>()=>{},
    totalPages: 10
  };

  const configs = [
    {
      props: mockProps,
      desc: 'should render normal PaginationBar'
    }
  ];

  assertSnapshots(PaginationBar, configs);
});
