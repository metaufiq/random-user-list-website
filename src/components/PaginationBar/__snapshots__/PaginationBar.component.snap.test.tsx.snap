// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationBarSnapshot should render normal PaginationBar 1`] = `
<nav
  aria-label="Page navigation example"
>
  <ul
    className="inline-flex -space-x-px"
  >
    <li>
      <button
        className="py-2 px-3 ml-0 leading-tight text-gray-500 bg-white rounded-l-lg border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        Previous
      </button>
    </li>
    <li>
      <button
        aria-current="page"
        className="py-2 px-3 text-blue-600 bg-blue-50 border border-gray-300 hover:bg-blue-100 hover:text-blue-700 dark:border-gray-700 dark:bg-gray-700 dark:text-white"
        onClick={[Function]}
      >
        1
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        2
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        3
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        4
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        5
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        6
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        7
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        8
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        9
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        10
      </button>
    </li>
    <li>
      <button
        className="py-2 px-3 leading-tight text-gray-500 bg-white rounded-r-lg border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        onClick={[Function]}
      >
        Next
      </button>
    </li>
  </ul>
</nav>
`;
